src/Compilers/CSharp/Portable/Syntax/SyntaxFactory.cs

```diff
@@ -2529,23 +2529,81 @@ public static ArgumentSyntax Argument(ExpressionSyntax expression)
         /// <summary>Creates a new ArgumentSyntax instance.</summary>
         public static ArgumentSyntax Argument(NameColonSyntax nameColon, SyntaxToken outKeyword, VariableDeclarationSyntax declaration)
         {
-            if (outKeyword.Kind() != SyntaxKind.OutKeyword)
+            return Argument(nameColon, outKeyword, (CSharpSyntaxNode)declaration);
+        }
+
+        /// <summary>Creates a new ArgumentSyntax instance.</summary>
+        public static ArgumentSyntax Argument(SyntaxToken outKeyword, VariableDeclarationSyntax declaration)
+        {
+            return Argument(default(NameColonSyntax), outKeyword, declaration);
+        }
+
+        static partial void ValidateArgumentParts(NameColonSyntax nameColon, SyntaxToken refOrOutKeyword, CSharpSyntaxNode expressionOrDeclaration)
```

Comment:
It is called from auto-generated
```
internal static ArgumentSyntax Argument(NameColonSyntax nameColon, SyntaxToken refOrOutKeyword, CSharpSyntaxNode expressionOrDeclaration)
```

after all standard auto-generated validations are performed.